pkg/model/network.go

```diff
@@ -148,35 +151,74 @@ func (b *NetworkModelBuilder) Build(c *fi.ModelBuilderContext) error {
 		}
 	}
 
-	for _, zone := range privateZones.List() {
+	for i, zone := range privateZones.List() {
+
 		utilitySubnet, err := b.LinkToUtilitySubnetInZone(zone)
 		if err != nil {
 			return err
 		}
 
+		// Has an existing NgwId been entered?
+		// NGWs look like this: ngwId: nat-09c4180b76a36ca2c
+		ngwId := b.Cluster.Spec.Subnets[i].NgwId
+
+		// Was an elasticIp also allocated? This needs to be the ElasticIP
+		// associated with the NAT Gateway ngwEips look like: ngwEip: eipalloc-e1fc20df
+		ngwEip := b.Cluster.Spec.Subnets[i].NgwEip
+
+		// If these get triggered, something has gone wrong in pkg/apis/kops/validation.go
+		if ngwId != "" && ngwEip == "" {
+			return fmt.Errorf("must specify the associated ElasticIP when specifying NAT Gateways")
+		}
+
+		if ngwEip != "" && ngwId == "" {
+			return fmt.Errorf("must specify a NAT Gateway when specifying ElasticIP")
+		}
+
 		// Every NGW needs a public (Elastic) IP address, every private
 		// subnet needs a NGW, lets create it. We tie it to a subnet
 		// so we can track it in AWS
-		eip := &awstasks.ElasticIP{
-			Name: s(zone + "." + b.ClusterName()),
-			AssociatedNatGatewayRouteTable: b.LinkToPrivateRouteTableInZone(zone),
+		var eip = &awstasks.ElasticIP{}
+		if ngwEip == "" {
+			eip = &awstasks.ElasticIP{
+				Name: s(zone + "." + b.ClusterName()),
+				AssociatedNatGatewayRouteTable: b.LinkToPrivateRouteTableInZone(zone),
+			}
+
+		} else {
+			eip = &awstasks.ElasticIP{
+				Name: s(zone + "." + b.ClusterName()),
+				AssociatedNatGatewayRouteTable: b.LinkToPrivateRouteTableInZone(zone),
+				ID: s(ngwEip),
```

Comment:
I think we likely want to set `Shared: true` as well, but maybe not because we're not going to tag it...  We just have to be careful when it comes to deletion time.